### PR TITLE
Run an actual job via the job runner

### PR DESCRIPTION
**Pull request by bloodearnest**
This involved actually running docker inside the test containers, which
was fidly.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -40,3 +40,8 @@ COPY packages.txt /tmp/packages.txt
 RUN sed 's/^#.*//' /tmp/packages.txt | xargs apt-get install -y
 
 RUN apt-get autoremove -y
+
+# this allows use to avoid docker in docker
```
Comment (madwort) [inline Dockerfile:44]: nitpick typo `use` -> `us`

File: scripts/jobrunner.sh [hunk 2]
```diff
@@ -9,6 +9,7 @@ umask 027
 
 # ensure shared user is set up properly
 id -u jobrunner >/dev/null 2>&1 || useradd jobrunner --create-home --shell /bin/bash -G docker
+usermod -a -G docker jobrunner
```
Comment (madwort) [inline scripts/jobrunner.sh:12]: not sure why we need this, `useradd` already has `-G docker` that should do the same thing?

**Review by madwort — CHANGES_REQUESTED** [review]
nits to check before merge

File: scripts/jobrunner.sh (hunk 2 above)
Comment (bloodearnest) [reply scripts/jobrunner.sh:12]: so it does!

Comment (bloodearnest) [reply scripts/jobrunner.sh:12]: ah I remember why I did this - some existing backends may have had their `jobrunner` user created without the `docker` group being added, so this fixes them

Comment (madwort) [reply scripts/jobrunner.sh:12]: ah you're right! if the `jobrunner` user already exists.

**Review by madwort — APPROVED** [review]

**Pull request merged**